### PR TITLE
Added docker-compose.yml

### PR DESCRIPTION
**Pull request by hueyy**
* Standard redis:alpine container
* Added instructions in README.md stating how to use docker-compose to run the API locally

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  redis:
+    image: redis:alpine
+    hostname: redis
+    ports:
+      - '127.0.0.1:6379:6379'
+    networks:
+      - redis-net
+
+networks:
```
Comment (ChristopherHammond13) [inline docker-compose.yml]: Is a separate network needed, or does the default overlay network do what's needed? Just wondering what the motivation here was.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+version: '3'
+
+services:
+  api:
+    build: .
+    image: ucl-assistant-api
+    env_file: .env
+    environment:
+      - REDIS_URL=redis://redis:6379?ConnectTimeout=5000&IdleTimeOutSecs=180
```
Comment (ChristopherHammond13) [inline docker-compose.yml]: Are these extra arguments needed in prod too?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  redis:
+    image: redis:alpine
+    hostname: redis
+    ports:
+      - '127.0.0.1:6379:6379'
+    networks:
+      - redis-net
+
+networks:
+  redis-net:
```
Comment (ChristopherHammond13) [inline docker-compose.yml]: `ucl-assistant` might be a better name for the local network, but it's not that much of a big deal 🤷‍♂️

**Comment by ChristopherHammond13** [comment]
Just a couple of questions/comments then I'm happy to merge. Good idea getting this in place!

**Comment by hueyy** [comment]
Fair points. I've cleaned it up a bit.

**Pull request merged**